### PR TITLE
[WIP] Start adding more injections

### PR DESCRIPTION
**Pull request by mvdoc**
Starting to add more -- shall we also add specific functions?

### CODE REVIEW

**Comment by yarikoptic** [comment]
YES -- specific functions is what we should aim for in particular. E.g. starting with scipy
Also, while at it, would be great if for every function you could deduce when that "method" was added (checkout changelog and file history in git) and add `min_version`?

**Comment by yarikoptic** [comment]
For numpy we will figure out what should "trigger" really its citation, although it is pretty much always needs to be there since nothing sciency could be done without it ;)  So may be we would need a flag (just a tag?) at the end for module level to signal that import of the module is sufficient to trigger its referencing... but that could be done later

**Comment by mvdoc** [comment]
OK cool. I think that going randomly to find stuff to inject is quite unproductive. Better if we had in mind a use case, and then I'll fill in with whatever we need. Do you have any ideas? Maybe some of the sklearn examples?

**Comment by yarikoptic** [comment]
Actually might be quite easy 
 git grep -A10 References
In local scipy git 
and output might give you ideas for use cases at once

Sklearn is just full of those :-)

File: duecredit/injections/mod_scipy.py [hunk 1]
```diff
@@ -25,4 +25,130 @@ def inject(injector):
...
+        title={Hierarchical clustering schemes},
+        author={Johnson, Stephen C},
+        journal={Psychometrika},
+        volume={32},
+        number={3},
+        pages={241--254},
+        year={1967},
+        publisher={Springer}
+    }"""), min_version='v0.4.3',
+                 description="Global references for the module")
```
Comment (yarikoptic) [inline duecredit/injections/mod_scipy.py]: See -- that is how "description" is probably misleading. Before it was "use", with original intent to signal to describe how this functionality is usually used.  E.g. here I probably would have said

```description="Hierarchical clustering", tags=["edu"]```

File: duecredit/injections/mod_scipy.py [hunk 2]
```diff
@@ -25,4 +25,130 @@ def inject(injector):
...
+    @article{johnson1967hierarchical,
+        title={Hierarchical clustering schemes},
+        author={Johnson, Stephen C},
+        journal={Psychometrika},
+        volume={32},
+        number={3},
+        pages={241--254},
+        year={1967},
+        publisher={Springer}
+    }"""), min_version='v0.4.3',
```
Comment (yarikoptic) [inline duecredit/injections/mod_scipy.py]: "v" here is their scheme on how to tag releases.  For our purpose, we want just version (i.e. "0.4.3")

File: duecredit/injections/mod_scipy.py (hunk 1 above)
Comment (yarikoptic) [inline duecredit/injections/mod_scipy.py]: entries below should also get tags

File: duecredit/injections/mod_scipy.py [hunk 3]
```diff
@@ -25,4 +25,130 @@ def inject(injector):
...
+        author={Ward Jr, Joe H},
+        journal={Journal of the American statistical association},
+        volume={58},
+        number={301},
+        pages={236--244},
+        year={1963},
+        publisher={Taylor \& Francis}
+    }"""), conditions={(1, 'method'): {'ward'}},
+                 description="Ward hierarchical clustering",
+                 min_version='v0.4.3')
```
Comment (yarikoptic) [inline duecredit/injections/mod_scipy.py]: I guess this is an example for ```tags=['implementation']``` which is the default, thus ok to not have explicitely listed here

**Comment by mvdoc** [comment]
I added the example because I need to leave now, however when I run it from command line with 
`DUECREDIT_ENABLE=True; python example_scipy.py` it doesn't create any `.duecredit.p` in the folder, so `duecredit summary` doesn't generate anything. Will look into it on the plane, but in the meanwhile I wanted to share the example with you.

**Comment by mvdoc** [comment]
After the fix in #17 this is what is being produced running the example_scipy.py.

```
2015-07-23 12:38:07,180 [DEBUG] Adding delayed injection for ('numpy',) (injector.py:73)
2015-07-23 12:38:07,180 [DEBUG] Adding delayed injection for ('scipy',) (injector.py:73)
2015-07-23 12:38:07,180 [DEBUG] Adding delayed injection for ('sklearn',) (injector.py:73)
2015-07-23 12:38:07,180 [DEBUG] Assigning our importer (injector.py:179)
2015-07-23 12:38:07,181 [Level 1] Module numpy.__config__ was imported (injector.py:172)
2015-07-23 12:38:07,181 [Level 1] Module version was imported (injector.py:172)
2015-07-23 12:38:07,181 [Level 1] Module version was imported (injector.py:172)
2015-07-23 12:38:07,181 [Level 1] Module _import_tools was imported (injector.py:172)
2015-07-23 12:38:07,182 [Level 1] Module info was imported (injector.py:172)
2015-07-23 12:38:07,183 [Level 1] Module info was imported (injector.py:172)
2015-07-23 12:38:07,184 [Level 1] Module  was imported (injector.py:172)
2015-07-23 12:38:07,184 [Level 1] Module  was imported (injector.py:172)
2015-07-23 12:38:07,184 [Level 1] Module  was imported (injector.py:172)
2015-07-23 12:38:07,185 [Level 1] Module  was imported (injector.py:172)
2015-07-23 12:38:07,185 [Level 1] Module _inspect was imported (injector.py:172)
2015-07-23 12:38:07,185 [Level 1] Module py3k was imported (injector.py:172)
2015-07-23 12:38:07,185 [Level 1] Module numpy.compat was imported (injector.py:172)
< ... ellipsis ...>
DueCredit Report:

0 modules cited
0 functions cited
```

**Comment by yarikoptic** [comment]
On Thu, 23 Jul 2015, Matteo Visconti di Oleggio Castello wrote:
>    DUECREDIT_ENABLE=True; python example_scipy.py 

remove ";" ;)  it makes it into two separate statements and since first
one is without export -- it doesn't propagate into any of the "child
processes"

**Comment by yarikoptic** [comment]
but that wasn't it -- I am looking into it. Cheers

**Pull request merged**